### PR TITLE
Validate label index

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: test/spectests/spectests.cpp [hunk 1]
```diff
@@ -289,6 +296,11 @@ class test_runner
                     pass();
```
Comment (axic) [inline test/spectests/spectests.cpp]: I think actually we should check the `type` here. It is a pass in `assert_malformed`, but a fail in `assert_invalid`.

File: test/spectests/spectests.cpp [hunk 2]
```diff
@@ -289,6 +296,11 @@ class test_runner
...
                     continue;
                 }
+                catch (fizzy::validation_error const&)
+                {
+                    pass();
```
Comment (axic) [inline test/spectests/spectests.cpp]: The inverse of https://github.com/wasmx/fizzy/pull/241/commits/32cc345a2b296e6625a1a3dcab3ec0a1bac84d24#r398071288 applies here.

Comment (chfast) [reply test/spectests/spectests.cpp]: This does not work.

File: test/spectests/spectests.cpp [hunk 3]
```diff
@@ -289,6 +296,15 @@ class test_runner
...
                     continue;
                 }
+                catch (fizzy::validation_error const&)
+                {
+                    if (type == "assert_invalid")
+                    {
+                        pass();
+                        continue;
+                    }
+                    throw;
```
Comment (gumb0) [inline test/spectests/spectests.cpp]: it should fail instead

Comment (gumb0) [reply test/spectests/spectests.cpp]: `throw` here just stops executing current file and outputs error to stderr

**Comment by gumb0** [comment]
Still need to hande `validation_error` in `assert_unlinkable`/`assert_uninstantiable`

File: test/spectests/spectests.cpp [hunk 4]
```diff
@@ -289,6 +296,14 @@ class test_runner
                     pass();
```
Comment (gumb0) [inline test/spectests/spectests.cpp]:
```suggestion
                    if (type == "assert_malformed")
                        pass();
                    else
                        fail(std::string{"Unexpected parser error: "} + ex.what());
                    continue;
```

**Pull request merged**